common/grpctracing/interceptor_test.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package grpctracing_test
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+
+	"github.com/hyperledger/fabric/common/grpctracing"
+	"github.com/hyperledger/fabric/common/grpctracing/fakes"
+	"github.com/hyperledger/fabric/common/grpctracing/testpb"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.opentelemetry.io/otel"
+	otelcodes "go.opentelemetry.io/otel/codes"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type TracesCollector struct {
+	spansCollected []sdktrace.ReadOnlySpan
+}
+
+func (t *TracesCollector) ExportSpans(ctx context.Context, spans []sdktrace.ReadOnlySpan) error {
+	t.spansCollected = append(t.spansCollected, spans...)
+	return nil
+}
+
+func (t *TracesCollector) Shutdown(ctx context.Context) error {
+	return nil
+}
+
+var _ = Describe("Interceptor", func() {
+	var (
+		tracesCollector   *TracesCollector
+		tracerProvider    *sdktrace.TracerProvider
+		fakeEchoService   *fakes.EchoServiceServer
+		echoServiceClient testpb.EchoServiceClient
+
+		listener        net.Listener
+		serveCompleteCh chan error
+		server          *grpc.Server
+	)
+
+	BeforeEach(func() {
+		tracesCollector = &TracesCollector{}
+		bsp := sdktrace.NewBatchSpanProcessor(tracesCollector)
+		tracerProvider = sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(bsp), sdktrace.WithSampler(sdktrace.AlwaysSample()))
+		otel.SetTracerProvider(tracerProvider)
+
+		var err error
+		listener, err = net.Listen("tcp", "127.0.0.1:0")
+		Expect(err).NotTo(HaveOccurred())
+
+		fakeEchoService = &fakes.EchoServiceServer{}
```

Comment:
how is the telemetry agent configured in a non-test case? I am failing to find it in this PR.

Reply:
This is done by using environment variables with the otlp trace exporter - see https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/otlp/otlptrace